### DISCUSSION TITLE
[minimal-blog] SEO with pathPrefix

### DISCUSSION THREAD

**Discussion by sh-cho** [LekoArts/gatsby-themes#1044] [Questions] [answered]
Hi, I've deployed blog on `https://joe-brothers.com` and set pathPrefix of gatsby with `/blog`.
- https://github.com/joe-brothers/blog

I know there's `blogPath` but if I set it, path is weird (like `joe-brothers.com/blog/blog/posts`

```js
module.exports = {
  pathPrefix: `/blog`,
  siteMetadata: {
    //..
    siteUrl: `https://joe-brothers.com/blog/`,
    siteImage: `/banner.jpg`,  // -> this is not working also.
    //..
  },
  plugins: [
    {
      resolve: `@lekoarts/gatsby-theme-minimal-blog`,
      // See the theme's README for all available options
      options: {
        // basePath: "/blog",
        blogPath: "/posts",
        formatString: "YYYY-MM-DD",
        navigation: [
          {
            title: `Posts`,
            slug: `/posts`,
          },
          {
            title: `Tags`,
            slug: `/tags`,
          },
        ],
        externalLinks: [
          {
            name: `Homepage`,
            url: `https://joe-brothers.com`,
          },
        ],
      },
    },
```
this is gatsby-config.js

```html
<meta name="image" content="https://joe-brothers.com/banner.jpg" data-gatsby-head="true"/>
<meta property="og:url" content="https://joe-brothers.com/test-post" data-gatsby-head="true"/>
<meta property="og:image" content="https://joe-brothers.com/static/2842f5bb607e08371d724f0ffba7311b/73f08/building.png" data-gatsby-head="true"/>
```
In seo urls, `/blog` path prefix is not included. (ex. https://joe-brothers.com/blog/static/2842f5bb607e08371d724f0ffba7311b/73f08/building.png -> This is right url)

I'm not sure what's wrong 😞.. Can you help me?

**Comment by LekoArts** [comment]
Hi, thanks for the question. In theory this should work like this, yes.

I'm taking the `siteUrl` and use that to construct the path. I'll need to check what's going on here: https://github.com/LekoArts/gatsby-themes/blob/main/themes/gatsby-theme-minimal-blog/src/components/seo.tsx

**Comment by LekoArts** [comment, marked as answer]
Mh, I just cloned the minimal blog starter, added a pathPrefix, changed the siteUrl and build the site with pathPrefix. And it works:

![image](https://user-images.githubusercontent.com/16143594/200180335-7a589cd6-b33e-4d91-a06d-2720c333cb8d.png)

Note that you have to build and serve with a CLI flag: https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/

**Reply by sh-cho** [reply]
https://github.com/joe-brothers/blog/blob/c5db261b96916ca7b579d88f87a1c9987b07958d/.github/workflows/gatsby.yml#L50

Thanks, but It was not the main problem. I find out that I need to remove [`actions/configure-pages`](https://github.com/actions/configure-pages) action's `static_site_generator` option if I want to manage pathPrefix myself.
